### ISSUE TITLE
Implement support for relationship on PATCH request

### ISSUE THREAD

**Issue by nohponex** [phramework/jsonapi#21]
See http://jsonapi.org/format/#crud-updating-resource-relationships

For instance, the following PATCH request will update the author relationship of an article:

```
PATCH /articles/1 HTTP/1.1
Content-Type: application/vnd.api+json
Accept: application/vnd.api+json

{
  "data": {
    "type": "articles",
    "id": "1",
    "relationships": {
      "author": {
        "data": { "type": "people", "id": "1" }
      }
    }
  }
}
```

Likewise, the following PATCH request performs a complete replacement of the tags for an article:

```
PATCH /articles/1 HTTP/1.1
Content-Type: application/vnd.api+json
Accept: application/vnd.api+json

{
  "data": {
    "type": "articles",
    "id": "1",
    "relationships": {
      "tags": {
        "data": [
          { "type": "tags", "id": "2" },
          { "type": "tags", "id": "3" }
        ]
      }
    }
  }
}
```

**Comment by tchronis** [comment]
A bit of code in the preview tab please :-)

**Comment by nohponex** [comment]
Implemented at [1.0.0-RC2](https://github.com/phramework/jsonapi/releases/tag/1.0.0-RC2)
